onelogin_aws_cli/argparse.py

```diff
@@ -0,0 +1,57 @@
+"""
+CLI Argument Parser
+"""
+
+import argparse
+
+import pkg_resources
+
+
+class OneLoginAWSArgumentParser(argparse.ArgumentParser):
+    """
+    Argument Parser separated into daemon and cli tool
+    """
+
+    def __init__(self):
+        super().__init__(description='Login to AWS with OneLogin')
+
+        self.add_argument(
+            '-C', '--config-name', default='default', dest='config_name',
+            help='Switch configuration name within config file'
+        )
+
+        self.add_argument(
+            '--profile', default='', help='Specify profile name of credential'
+        )
+
+        self.add_argument(
+            '-u', '--username', default='', help='Specify OneLogin username'
+        )
+
+    def add_cli_options(self):
+        """
+        Add Argument Parser options only used in the CLI entrypoint
+        """
+
+        self.add_argument(
+            '-r', '--renew-seconds', type=int,
+            help='Auto-renew credentials after this many seconds'
+        )
+
+        self.add_argument(
+            '-c', '--configure', dest='configure', action='store_true',
+            help='Configure OneLogin and AWS settings'
+        )
+
+        version = pkg_resources.get_distribution(__package__).version
```

Comment:
Should version be common to both?

Reply:
My test for each option was: "If I run this in daemon mode, will `insert option under consideration` stop the daemon from running?"

Although this would fail that little test, on reflection, `--version` feels special as it would be fair to ask "what is the daemon version?".